src/core/server/config/config.ts

```diff
@@ -75,3 +75,13 @@ export interface Config {
    */
   toRaw(): Record<string, any>;
 }
+
+const pathDelimiter = '.';
+export function hasConfigPathIntersection(leafPath: string, rootPath: string) {
+  if (!leafPath || !rootPath) {
+    throw new Error(`Paths were expected, but given leaf: ${leafPath} and root: ${rootPath}`);
```

Comment:
If these are empty, and since they must be strings, this error message will produce either of the following:

> Paths were expected, but given leaf:  and root: a.b

> Paths were expected, but given leaf: a.b and root: 

In order to make this more useful, let's split this into 2 errors:

```ts
if (!leafPath) {
  throw new Error('leafPath cannot be empty');
} else if (!rootPath) {
  throw new Error('rootPath cannot be empty');
}
```